tests/appsec/test_blocking_full_denylist.py

```diff
@@ -0,0 +1,82 @@
+from utils import context, released, coverage, interfaces, scenarios, weblog, bug, irrelevant, missing_feature
+
+
+@released(
+    cpp="?",
+    dotnet="2.30.0",
+    golang="1.48.0",
+    java="1.7.0",
+    nodejs="3.15.0",
+    php="0.85.0",
+    php_appsec="0.7.0",
+    python="1.12.1",
+    ruby="?",
+)
+@missing_feature(context.weblog_variant == "spring-boot-native", reason="GraalVM. Tracing support only")
+@missing_feature(context.weblog_variant == "spring-boot-3-native", reason="GraalVM. Tracing support only")
+@irrelevant(context.appsec_rules_file is not None)
+@bug(context.weblog_variant == "uds-echo")
+@bug(context.library > "nodejs@3.16.0", reason="Under investigation")
+@coverage.basic
+@scenarios.appsec_blocking_full_denylist
+class Test_BlockingFullDenylist:
+    def setup_nonblocking_test(self):
+        self.r_nonblock = weblog.get("/users", params={"user": "regularUser"})
+
+    def setup_blocking_test(self):
+        self.r_block = weblog.get("/users", params={"user": "999"})
+
+    def test_nonblocking_test(self):
+        NOT_BLOCKED_USER = "regularUser"
+
+        def validate_nonblock_user(span):
+            assert span["meta"]["usr.id"] == NOT_BLOCKED_USER
+            return True
+
+        assert self.r_nonblock.status_code == 200
+        interfaces.library.validate_spans(self.r_nonblock, validator=validate_nonblock_user)
+        interfaces.library.assert_no_appsec_event(self.r_nonblock)
+
+    def test_blocking_test(self):
+        """Test with a denylisted user"""
+
+        def validate_blocking_test(span):
+            """Check all fields are present in meta"""
+            for i in range(1250):
+                assert span["meta"]["usr.id"] == str(i)
```

Comment:
I may have missed something, but what you are doing is : 

```python
assert span["meta"]["usr.id"] == "0" and  span["meta"]["usr.id"] == "1"
```

So it can never be ok